### PR TITLE
wait after spanner test client is disconnected

### LINKED ISSUE #49495: `TestAuditSpanner/access_denied` flakiness
#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/12048114969/job/33592068420#step:6:2132

#### Relevant snippet

```
     spanner_test.go:254: 
        	Error Trace:	/__w/teleport/teleport/lib/srv/db/audit_test.go:423
        	            				/__w/teleport/teleport/lib/srv/db/audit_test.go:394
        	            				/__w/teleport/teleport/lib/srv/db/spanner_test.go:254
        	Error:      	timed out waiting for an audit event
        	Test:       	TestAuditSpanner/access_denied
        	Messages:   	didn't receive any event in time
```

### PR DESCRIPTION
**Pull request by GavinFrazar**
Fixes #49495
- #49495

### CODE REVIEW

File: lib/srv/db/spanner_test.go [hunk 1]
```diff
@@ -246,6 +246,10 @@ func TestAuditSpanner(t *testing.T) {
 		require.True(t, ok)
 		require.Equal(t, "googlesql", dbStart1.DatabaseName)
 
+		// the connection should start to shut down, but wait some time for that
+		// to happen before sending further RPCs so that the client will dial
+		// a new connection before sending another query RPC.
+		time.Sleep(3 * time.Second)
```
Comment (greedy52) [inline lib/srv/db/spanner_test.go:252]: any way to avoid the "long" sleep?

Comment (GavinFrazar) [reply lib/srv/db/spanner_test.go:252]: I couldn't reproduce locally with a 10ms sleep, so I'll reduce this

File: lib/srv/db/spanner_test.go [hunk 2]
```diff
@@ -246,6 +246,10 @@ func TestAuditSpanner(t *testing.T) {
 		require.True(t, ok)
 		require.Equal(t, "googlesql", dbStart1.DatabaseName)
 
+		// the connection should start to shut down, but wait some time for that
+		// to happen before sending further RPCs so that the client will dial
+		// a new connection before sending another query RPC.
+		time.Sleep(500 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (tigrato) [reply lib/srv/db/spanner_test.go:252]: You can then pull the state and match it with your desired state

File: lib/srv/db/spanner_test.go (hunk 1 above)
Comment (greedy52) [reply lib/srv/db/spanner_test.go]: or maybe we wait util server has no action connections?

File: lib/srv/db/spanner_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply lib/srv/db/spanner_test.go]: done https://github.com/gravitational/teleport/commit/53f912c7cce4ca8e77bf1ac2191034cac449320d

**Review by tigrato — APPROVED** [review]
Thanks. Much better now

**Review by greedy52 — APPROVED** [review]

**Pull request merged**